language_generators/scala-defref-extractor/src/main/scala/io/bazeltools/buildgen/scaladefref/ScalaSourceEntityExtractor.scala

```diff
@@ -761,16 +871,26 @@ object ScalaSourceEntityExtractor {
             ctor,
             template @ Template(early, init, self, stats)
           ) =>
-        log(s"Class($mods, $name, $tparams, $ctor, $template)")
-        define(name) *> inside(
+        log(
+          s"Class($mods, $name, $tparams, $ctor, Template(early: $early, init: $init, self:$self, stats: $stats)"
+        )
+
+        define(name) *> link(
+          name,
+          init.map(_.tpe).flatMap(typeToNames) ++ stats.flatMap(treeToNames)
+        ) *> inside(
           name, {
             (tparams ::: ctor :: template :: early ::: init ::: self :: Nil)
               .traverse_(inspect) *> stats.traverse_(inspect(_))
           }
         )
-      case Defn.Object(mods, name, templ) =>
+      case Defn.Object(mods, name, templ @ Template(_, init, _, stats)) =>
         log(s"Object($mods, $name, $templ)")
-        define(name) *> inside(name, inspect(templ))
+
+        define(name) *> link(
```

Comment:
so the only difference here is define(name) or not, right? Could we add a comment to remind us later?